server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregator.java

```diff
@@ -231,6 +235,10 @@ private Sort buildIndexSortPrefix(LeafReaderContext context) throws IOException
                 break;
             }
             sortFields.add(indexSortField);
+            if (sourceConfig.valuesSource() instanceof RoundingValuesSource) {
+                // rounded values break the index sort
```

Comment:
It's probably worth explaining why. I *think* it is because the rounding "squashes" many values together so we can't get the order of the sub-values by looking at any one key. We'd have to merge all the values that round to that key and get the sub-values from that. And that isn't the kind of this this index sorting optimization does.

Reply:
++, I pushed https://github.com/elastic/elasticsearch/pull/57867/commits/e48605775859aeb48d6f0eb1d7356ca434fdf6d2